### PR TITLE
Queue up anything support

### PR DESCRIPTION
**Pull request by stephenbinns**
This can be used to assert that any job was queued

```
expect { thing }.to_not queue_up(anything)
```

Previously to write this assertion we would have to write the following:

```
expect { thing }.to_not queue_up(nil)
```

Which is non intuitive, this fixes the matchers to allow any class to be used.

### CODE REVIEW

File: spec/rspec/que/queue_up_spec.rb [hunk 1]
```diff
@@ -24,6 +24,17 @@ class CJob; end
...
     end
+
+    context "when it enqueues a job and we allow anything" do
+      let(:job_class) { anything }
+
+      it { is_expected.to be(false) }
+      specify do
+        matches?
+        expect(instance.failure_message).
+          to eq("expected to enqueue a job any class, but found nothing")
```
Comment (cezarelnazli) [inline spec/rspec/que/queue_up_spec.rb:35]: Does this read better? 🤔

```suggestion
          to eq("expected to enqueue a job of any class, but found nothing")
```

Comment (stephenbinns) [reply spec/rspec/que/queue_up_spec.rb:35]: Yeah sounds good

File: lib/rspec/que.rb [hunk 2]
```diff
@@ -6,6 +8,10 @@ def queue_up(job_class)
       Matchers::QueueUp.new(job_class)
     end
 
+    def queue_up_anything
```
Comment (stephenbinns) [inline lib/rspec/que.rb:11]: Thinking about ditching this now @cezarelnazli as the new method is better

Comment (cezarelnazli) [reply lib/rspec/que.rb]: Yup, I think it's more in line with how RSpec works.

**Pull request merged**